src/test/librbd/test_mock_fixture.h

```diff
@@ -25,7 +25,11 @@ ACTION_P(CopyInBufferlist, str) {
 }
 
 ACTION_P2(CompleteContext, r, wq) {
+#if defined(__clang__)
+  ContextWQ *context_wq = static_cast<ContextWQ *>(wq);
```

Comment:
@dillaman 
Compiling GnuCC gives:
```
In file included from /home/wjw/wip/src/test/librbd/test_mock_Journal.cc:4:0:
/home/wjw/wip/src/test/librbd/test_mock_fixture.h: In instantiation of ‘typename testing::internal::Function<F>::Result CompleteContextActionP2<r_type, wq_type>::gmock_Impl<F>::gmock_PerformImpl(const args_type&, arg0_type, arg1_type, arg2_type, arg3_type, arg4_type, arg5_type, arg6_type, arg7_type, arg8_type, arg9_type) const [with arg0_type = Context*; arg1_type = testing::internal::ExcessiveArg; arg2_type = testing::internal::ExcessiveArg; arg3_type = testing::internal::ExcessiveArg; arg4_type = testing::internal::ExcessiveArg; arg5_type = testing::internal::ExcessiveArg; arg6_type = testing::internal::ExcessiveArg; arg7_type = testing::internal::ExcessiveArg; arg8_type = testing::internal::ExcessiveArg; arg9_type = testing::internal::ExcessiveArg; F = void(Context*); r_type = int; wq_type = long int; typename testing::internal::Function<F>::Result = void; CompleteContextActionP2<r_type, wq_type>::gmock_Impl<F>::args_type = std::tuple<Context*>]’:
/home/wjw/wip/src/googletest/googlemock/include/gmock/gmock-generated-actions.h:526:23:   required from ‘static Result testing::internal::ActionHelper<Result, Impl>::Perform(Impl*, const std::tuple<A0>&) [with A0 = Context*; Result = void; Impl = CompleteContextActionP2<int, long int>::gmock_Impl<void(Context*)>]’
/home/wjw/wip/src/test/librbd/test_mock_fixture.h:27:1:   required from ‘CompleteContextActionP2<r_type, wq_type>::gmock_Impl<F>::return_type CompleteContextActionP2<r_type, wq_type>::gmock_Impl<F>::Perform(const args_type&) [with F = void(Context*); r_type = int; wq_type = long int; CompleteContextActionP2<r_type, wq_type>::gmock_Impl<F>::return_type = void; CompleteContextActionP2<r_type, wq_type>::gmock_Impl<F>::args_type = std::tuple<Context*>]’
/home/wjw/wip/src/test/librbd/test_mock_Journal.cc:1349:1:   required from here
/home/wjw/wip/src/test/librbd/test_mock_fixture.h:32:54: error: invalid static_cast from type ‘const long int’ to type ‘ContextWQ*’
   ContextWQ *context_wq = static_cast<ContextWQ *>(wq);
```

Which is probably again due to the way that in linux NULL is defined...